### PR TITLE
refactor: MyMeta profile using react-grid-layout

### PR DESCRIPTION
**Pull request by dan13ram**
## Overview

**What features/fixes does this PR include?**

This PR is working towards "Add New Block" functionality for MyMeta profiles. This PR adds a new implementation for the MyMeta profile page using the `react-grid-layout` library which is also being used on the Dashboard page. 
With this change the player can drag and resize different sections of their profile.

**Please provide the GitHub issue number**

This works towards #264

## Follow up Improvement Ideas

- [ ] Support new section types 

**Side effects**

- Default height of different sections on the profile page displays empty space for players who've not updated their profile.
- The above is the result of fixed width and height of each section as mandated by `react-grid-layout`.

## Assets

Here is how it looks when editing the layout.
Screenshot from 2022-01-04 06:01:14

Add new section:
Screenshot from 2022-01-04 06:01:33
Screenshot from 2022-01-04 06:02:02
Screenshot from 2022-01-04 06:02:51

Dragging and Resizing:
Screenshot from 2022-01-04 06:02:51
Screenshot from 2022-01-04 06:03:03

### CODE REVIEW

File: packages/design-system/src/icons/Profile.tsx [hunk 1]
```diff
@@ -7,16 +7,16 @@ export const Profile = createIcon({
     <g>
       <g>
         <path
-          fill-rule="evenodd"
-          clip-rule="evenodd"
+          fillRule="evenodd"
+          clipRule="evenodd"
```
Comment (vidvidvid) [inline packages/design-system/src/icons/Profile.tsx:11]: thanks for fixing that :)

**Comment by peth-yursick** [comment]
Great to see progress on this! Thank you 🙏

**Comment by vidvidvid** [comment]
Really nice, I tested the layout editing and looked through the code and couldn't find any suggestions for improvements. Imo we could ship this as it is! 🚀

**Comment by dan13ram** [comment]
Thanks @vidvidvid 
What do you think ? Shall I replace the `pages/player/[username].tsx` with `pages/player/grid/[username].tsx' file ?

**Comment by vidvidvid** [comment]
> Thanks @vidvidvid What do you think ? Shall I replace the `pages/player/[username].tsx` with `pages/player/grid/[username].tsx` file ?

sure, why not! could you in this case please make sure that all of the profile sections keep the default sizes? when i checked the `/grid` layout the sizes of some sections were unnecessarily large.

**Comment by vidvidvid** [comment]
also - as discussed on discord, please:
1. hide the edit button until the persistence of layout is implemented
2. figure out how to dynamically adapt the layout to content

How it is atm:
Screenshot 2022-01-05 at 16 22 24

How it should be:
Screenshot 2022-01-05 at 16 24 27

**Review by luxumbra — COMMENTED** [review]
A great upgrade to the profiles! Would be good to get some of this into the Dashboard to update some of the styling and also WRT storing the layout data. I used local storage as a quick solution for the dashboard but if the layout here is getting a home in hasura, maybe the dashboard can share that too. 
image

One issue i found with `react-grid-layout` was that the *heights* of sections, when changed and saved, wouldn't position correctly on refresh. Did you have issues with that?

**Comment by dan13ram** [comment]
@luxumbra Thank you kindly for your help in implementing the "Embedded Url" Section. 
I've added that to the grid layout setup now.

> One issue i found with `react-grid-layout` was that the heights of sections, when changed and saved, wouldn't position correctly on refresh. Did you have issues with that?

Regarding this, I am facing the same issue if I try to persist the layout data in localStorage and retrieve it. I've an feeling that it could have something to do with SSR. I'm gonna try persisting it in the database and check if the issue persists

**Comment by dan13ram** [comment]
@vidvidvid I've fixed some of the issues you've mentioned. Having implemented persitence in a new PR #1029 I'm leaving the edit button visible in this PR. 

> figure out how to dynamically adapt the layout to content

This is non-trivial and I shall deal with it in a separate PR.

**Comment by luxumbra** [comment]
> @luxumbra Thank you kindly for your help in implementing the "Embedded Url" Section.
> I've added that to the grid layout setup now.

I just tried it out. 😃 I must say, it is great to see that bit of functionality finally, after many months get added to the profiles! Nice job all round on this chunk of work and great to see you back on the buidl on MyMeta dan. ⚔️ 

I did notice that on my display, the box with the text after the image in the Chakra LinkBox has a ~20px gap at the base of the section. I think the bg color and box shadow on that element are leftovers from when i worked on it and could be dropped. 
image

> I'm gonna try persisting it in the database and check if the issue persists, perhaps in another PR.

Glad it wasn't just me that had this issue. Fingers crossed it is more reliable when using the db. 🤞🏻

**Comment by dan13ram** [comment]
Changed this to draft and rebased with #1029 .. We can test this with persistance when that PR is merged :)

**Comment by dan13ram** [comment]
@luxumbra @vidvidvid 
I've added persistance to hasura and fixed the ui bugs except for dynamically resizing the grid box to the content.

**Comment by vidvidvid** [comment]
I've tested the layout and apart from being really happy that you made it work, I have a couple of observations:

1. The positioning of the profile hero has moved

Before:
Screenshot 2022-01-10 at 21 45 05
After:
Screenshot 2022-01-10 at 21 45 10

2. After playing around with layout, saving it and refreshing the page - if i attempted to reset layout I assumed i'd get the default layout, but the layout changed to the last saved version. Would it make sense to add another button with the default layout?

For the sake of mvp i think the current state of this PR is sufficient, but i'd like to ask you to create two issues from the mentioned observations (if you won't address these issues in this PR) and another issue for dynamically determining the most fitting default height of player sections. Probably these could be tagged with `good first issue`.

**Comment by dan13ram** [comment]
@vidvidvid Thank you for your kind feedback.

1) I changed it so in order to give more room for creativity to the player. Now they have the option to move the PlayerHero section of their profile. Also moving the section with a negative margin gave some glitches in `react-grid-layout`. So I decided to do this. imo if this is not a deal breaker, we should keep it this way. 

2) This can be easily done. 

I've created issues for dealing with 2 and dynamic resizing of the grid items. Let me know if this is good to merge now :)

**Comment by vidvidvid** [comment]
> I've created issues for dealing with 2 and dynamic resizing of the grid items. Let me know if this is good to merge now :)

Sure, let's merge!

**Review by vidvidvid — APPROVED** [review]
Hmm, does github require explicit approval of a pull request in order for it to be merged? If so - I'm sorry for not doing it earlier x).

**Comment by dan13ram** [comment]
@vidvidvid Yes, thats right. We have setup our repo to allow merging of PRs only if there is atleast one approval

**Pull request merged**